### PR TITLE
Allow context_from_plack_request to bubble up custom HTTP responses.

### PR DESCRIPTION
**Pull request by wolfsage**
For example, if it fails to find a session, it may want to throw a
401 response.

### CODE REVIEW

File: t/lib/Bakesale/Context.pm [hunk 1]
```diff
@@ -51,4 +51,25 @@ package Bakesale::Context::System {
...
 }
 
+package Bakesale::Context::NoAuth {
+  use Moose;
+
+  sub code { 401 }
+
+  with 'Ix::Context::Error';
+}
+
```
Comment (wolfsage) [inline t/lib/Bakesale/Context.pm]: I figure most cases will look like the above - we want to return a specific http response code and nothing more

File: t/lib/Bakesale/Context.pm [hunk 2]
```diff
@@ -51,4 +51,25 @@ package Bakesale::Context::System {
...
+package Bakesale::Context::BadAuth {
+  use Moose;
+  use experimental qw(lexical_subs signatures postderef);
+
+  sub code { 401 }
+
+  sub modify_response ($self, $resp) {
+    $resp->body('{"error":"bad auth"}');
+  }
+
```
Comment (wolfsage) [inline t/lib/Bakesale/Context.pm]: This case is if we want to specify a specific body, or more likely, nuke the user's cookies

File: lib/Ix/App.pm [hunk 3]
```diff
@@ -60,7 +62,22 @@ sub _build_psgi_app ($self) {
       ];
     }
 
-    my $ctx = $self->processor->context_from_plack_request($req);
+    my ($ctx, $exception);
+
+    try {
```
Comment (rjbs) [inline lib/Ix/App.pm]: Discussed on Slack.  This should be moved up into App's general "big eval wrapper" so the "handle HTTP::Exceptions?" comment becomes code.

**Review by rjbs — CHANGES_REQUESTED** [review]
Looks good, modulo the one change called out and discussed.

**Review by rjbs — APPROVED** [review]

**Pull request merged**